src/acl/agent.test.ts

```diff
@@ -848,6 +851,540 @@ describe("getAgentResourceAccessModesAll", () => {
   });
 });
 
+describe("setAgentResourceAccessModes", () => {
+  it("adds Quads for the appropriate Access Modes", () => {
+    const sourceDataset = Object.assign(
+      getMockDataset("https://arbitrary.pod/resource.acl"),
+      { accessTo: "https://arbitrary.pod/resource" }
+    );
+
+    const updatedDataset = unstable_setAgentResourceAccessModes(
+      sourceDataset,
+      "https://some.pod/profileDoc#webId",
+      {
+        read: true,
+        append: true,
```

Comment:
Yeah, Michiel kept saying that it was implied and that we shouldn't add Append access if we add Write, but I can't actually find it more explicit than this:

> gives a more limited ability to write to a resource

Sort-of implying that it is a subset of Write, but not too explicit.

What we currently do:

- If an ACL says Write, we also report `append: true` (even if it does not say Append).
- We do not write Append if `write: true`.
- We make TypeScript yell at you if you try to set `append: false` and `write: true`.

The first two seem consistent, but there's something to be said for making them both do the opposite. I'll bring that up with the relevant panel too.

Reply:
@NSeydoux Hmm, so the ACL vocab says:

```
    acl:Append     a :Class;
         :comment """Append accesses are specific write access which only add information, and do not remove information.
    For text files, for example, append access allows bytes to be added onto the end of the file.
    For RDF graphs, Append access allows adds triples to the graph but does not remove any.
    Append access is useful for dropbox functionality.
    Dropbox can be used for link notification, which the information added is a notification
    that a some link has been made elsewhere relevant to the given resource.
    """;
         :label "append"@en;
         :subClassOf acl:Access,
                acl:Write .
```

Most importantly: `:subClassOf acl:Write`.

I'd have expected that to be the other way around, but obviously I'm not an RDF expert :P So to verify: I presume that that says that `acl:Write` _does_ entail `acl:Append`?

Reply:
Well that's a surprise, I wouldn't have expected things to be modeled this way ^^. I find it a little counter-intuitive, because It means that if `:a rdf:type acl:Append`, then `:a rdf:type acl:Write` is also true, but if you look at it with a set-oriented mindset, it means that the set of `Write` access contains the set of `Append` access, and the set of `Append` access is smaller than the set of `Write` access. Since we're not typing our rules as `acl:Append` or `acl:Write`, the set-oriented way of thinking better captures the intent i think, and indeed `acl:Write` entails `acl:Append`.

Reply:
Hmm yeah, still hard to wrap my mind around :P But that means that this is Resolved, right: we don't need to add Append if we add Write?